src/merkle_sum_tree/utils/hash.rs

```diff
@@ -1,33 +1,34 @@
-use crate::chips::poseidon::spec_entry::MySpec as PoseidonSpecEntry;
-use crate::chips::poseidon::spec_node::MySpec as PoseidonSpecNode;
-use crate::merkle_sum_tree::{R_L_ENTRY, R_L_NODE, WIDTH_ENTRY, WIDTH_NODE};
+use crate::chips::poseidon::poseidon_spec::PoseidonSpec;
+use crate::merkle_sum_tree::{L_ENTRY, L_NODE};
 use halo2_gadgets::poseidon::primitives::{self as poseidon, ConstantLength};
 use halo2_proofs::halo2curves::bn256::Fr as Fp;
 
+const WIDTH: usize = 7;
+const RATE: usize = 6;
+
 pub fn poseidon_node<const N_ASSETS: usize>(
     l1: Fp,
     l2: [Fp; N_ASSETS],
     r1: Fp,
     r2: [Fp; N_ASSETS],
 ) -> Fp {
-    let mut hash_inputs: [Fp; R_L_NODE] = [Fp::zero(); R_L_NODE];
+    let mut hash_inputs: [Fp; L_NODE] = [Fp::zero(); L_NODE];
 
     hash_inputs[0] = l1;
     hash_inputs[1..N_ASSETS + 1].copy_from_slice(&l2);
     hash_inputs[N_ASSETS + 1] = r1;
     hash_inputs[N_ASSETS + 2..2 * N_ASSETS + 2].copy_from_slice(&r2);
 
-    poseidon::Hash::<Fp, PoseidonSpecNode, ConstantLength<R_L_NODE>, WIDTH_NODE, R_L_NODE>::init()
+    poseidon::Hash::<Fp, PoseidonSpec, ConstantLength<L_NODE>, WIDTH, RATE>::init()
         .hash(hash_inputs)
 }
 
 pub fn poseidon_entry<const N_ASSETS: usize>(left: Fp, right: [Fp; N_ASSETS]) -> Fp {
```

Comment:
I agree on this modification. 
Just for the record, before we were using 2 different Poseidon Hasher.
- 1 for hashing the entry which is an array `[username, balance#1, balance#2]` considering the case of 2 Assets
- 1 for hashing the node which is an arrat `[hashLeft, balance#1Left, balance#2Left, hashRight, balance#1Right, balance#2Right]`

The previous entry hasher was smaller in terms of RATE and WIDTH and therefore more efficient. But I think that this way makes it more simple.